### PR TITLE
Update serverless docs to use qiskit patterns

### PR DESCRIPTION
**Pull request by psschwei**
cc @jenglick

### CODE REVIEW

**Comment by psschwei** [comment]
This is still pending final verification that the updated script works as expected, so please hold off on merging until Jen approves (don't see a way for me to mark as a draft or add a label)

**Comment by jenglick** [comment]
@javabster This is ready to merge when you are

**Comment by psschwei** [comment]
link failure is unrelated to changes made in this PR


> ❌ docs/api/qiskit/qiskit.transpiler.passes.LookaheadSwap.md: Failed to fetch 'https://medium.com/qiskit/improving-a-quantum-compiler-48410d7a7084': fetch failed
>
>Some links appear broken 💔

**Comment by Eric-Arellano** [comment]
> link failure is unrelated to changes made in this PR

Pardon that. I had a similar failure in my PR. I'll restart CI. 

The external link checker is new, so we're still working out the kinks.

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:6]: Can we go back to "workloads" for just this section title and also the above page title and description? Since "Qiskit Pattern" isn't yet introduced and isn't a widely known concept yet, it may be better to use a more familiar word for the start of this page

**Comment by psschwei** [comment]
Lint check may need to be rekicked (failure seems unrelated to this PR)

image

File: docs/run/quantum-serverless.mdx
Comment (javabster) [inline docs/run/quantum-serverless.mdx:17]: I think this heading could be a bit more descriptive, maybe something like:

```suggestion
## Qiskit Patterns with Quantum Serverless
```

File: docs/run/quantum-serverless.mdx
Comment (javabster) [inline docs/run/quantum-serverless.mdx:34]:
```suggestion
## Build a Qiskit Pattern
```
across the docs we try to avoid -ing tense

File: docs/run/quantum-serverless.mdx
Comment (javabster) [inline docs/run/quantum-serverless.mdx:71]:
```suggestion
## Deploy a Qiskit Pattern to Quantum Serverless
```

File: docs/run/quantum-serverless.mdx
Comment (javabster) [inline docs/run/quantum-serverless.mdx:91]:
```suggestion
## Runn a Qiskit Pattern remotely on Quantum Serverless
```

File: docs/run/quantum-serverless.mdx
Comment (abbycross) [inline docs/run/quantum-serverless.mdx:10]:
```suggestion
Try out the tutorials in [IBM Quantum Learning](https://learning.quantum.ibm.com/catalog/tutorials?topics=qiskit-patterns) and explore more of the features of Quantum Serverless in the [documentation](https://qiskit-extensions.github.io/quantum-serverless/).
```

File: docs/run/quantum-serverless.mdx
Comment (abbycross) [inline docs/run/quantum-serverless.mdx:28]:
```suggestion
## Qiskit Patterns with Quantum Serverless

Creating utility-scale quantum applications generally requires a variety of compute resource requirements. You can use Quantum Serverless to easily submit quantum workflows for remote, managed execution. These quantum workflows can typically be implemented within a common pattern, called a Qiskit Pattern. A Qiskit Pattern is an intuitive, repeatable set of steps for implementing a quantum computing workflow.

Steps in a Qiskit Pattern:

1. Map classical inputs to a quantum problem
2. Optimize problem for quantum execution
3. Execute using Qiskit Runtime primitives
4. Post-process, return result in classical format

Once you have built a Qiskit Pattern, you can use Quantum Serverless to deploy it and submit it for managed execution. Overall, the process of creating quantum software and submitting it for managed execution on a remote cluster can be broken down into three steps:
```

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:10]: Right after the quantum learning link, we should add a parenthetical note along the lines of:

> Try out the tutorials in [IBM Quantum Learning](https://learning.quantum.ibm.com/catalog/tutorials?topics=qiskit-patterns) (note: these are accessible in the Premium Plan once you have logged into your IBM Quantum account) and explore more of the features of Quantum Serverless in the [open-source documentation](https://qiskit-extensions.github.io/quantum-serverless/).

**Review by jenglick — COMMENTED** [review]
I'd like to include a note about the resource management alpha release here. Perhaps a subsection at the end of this page with:

```
## Resource Management (Alpha)

Premium Plan users have access to an alpha release of resource management functionality through Quantum Serverless. This enables automatic selection of quantum hardware for your workloads. A how-to guide is available on [IBM Quantum Learning](https://learning.quantum.ibm.com/tutorial/resource-management-automatically-select-quantum-hardware).
```

cc @javabster @abbycross

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:23]: step 1 is now "Generate quantum circuits and operators." (generate instead of map)

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:31]: to match later subsection title: "Deploy to Quantum Serverless"

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:32]: "Run remotely on Quantum Serverless"

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:50]: `# Step 1: Generate quantum circuits and operators.`

File: docs/run/quantum-serverless.mdx
Comment (abbycross) [inline docs/run/quantum-serverless.mdx:109]:
```suggestion
## Resource management (alpha)
```

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:23]: Apologies -- we're back to "map" here:

> Map classical inputs to a quantum problem

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:50]: back to "map"
```
# Step 1: Map classical inputs to a quantum problem
```

**Review by jenglick — COMMENTED** [review]
@psschwei sorry for the switch back to "map"

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:6]: Here and in the above "title" and "description" lines can we add "remotely":
```
# Run workloads remotely with Quantum Serverless
```

File: docs/run/quantum-serverless.mdx
Comment (jenglick) [inline docs/run/quantum-serverless.mdx:30]: For these three steps (build, deploy, run), can we enumerate them with 1, 2, 3

**Review by javabster — APPROVED** [review]
LGTM 🚀

**Pull request merged**
